ldc2 with -g flag fails to link on Apple M1 Pro (aarch64) with unaligned pointer

I removed everything I could think of:

```console
% cat testit.d
extern(C) void main() {}
% ldc2 -betterC testit.d
% ./testit              
% ldc2 -betterC -g testit.d
ld: warning: pointer not aligned at address 0x10000402D (anon + 45 from testit.o)
ld: unaligned pointer(s) for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Error: /usr/bin/cc failed with status: 1
```

> I sadly don't have too much time for LDC development these days, but feel free to ping me if you need any testing as well – just got a M1 Max/Monterey box.

I now too have an M1 Macbook Air, so can test too. 

**Off topic**: Already ran into troubles of course :). For now, I managed to build LDC (arm64 binary) by setting `MACOSX_DEPLOYMENT_TARGET=11` and using the LDC 1.27.0 arm64 released binary as bootstrapping compiler. Running cmake the first time, it errors with
```
CMake Error at cmake/Modules/ExtractDMDSystemLinker.cmake:42 (message):
  Failed to link empty D program using
  '/Users/johan/dcompilers/ldc-1.27.1/bin/ldmd2 -wi -link-debuglib':

  ld: library not found for -lpthread

  clang: error: linker command failed with exit code 1 (use -v to see
  invocation)

  Error: /Library/Developer/CommandLineTools/usr/bin/cc failed with status: 1
Call Stack (most recent call first):
  CMakeLists.txt:670 (include)
```
simply rerunning cmake "fixes" it, and happily finishes.

--- a/gen/dibuilder.cpp
+++ b/gen/dibuilder.cpp
@@ -368,8 +368,6 @@ DIType DIBuilder::CreateEnumType(TypeEnum *type) {
 }
 
 DIType DIBuilder::CreatePointerType(TypePointer *type) {
-  llvm::Type *T = DtoType(type);
-
   // TODO: The addressspace is important for dcompute targets. See e.g.
   // https://www.mail-archive.com/dwarf-discuss@lists.dwarfstd.org/msg00326.html
   const llvm::Optional<unsigned> DWARFAddressSpace = llvm::None;
@@ -378,9 +376,7 @@ DIType DIBuilder::CreatePointerType(TypePointer *type) {
 
   return DBuilder.createPointerType(
       CreateTypeDescription(type->nextOf(), /*voidToUbyte=*/true),
-      getTypeAllocSize(T) * 8, // size (bits)
-      getABITypeAlign(T) * 8,  // align (bits)
-      DWARFAddressSpace, name);
+      target.ptrsize * 8, 0, DWARFAddressSpace, name);
 }
 
 DIType DIBuilder::CreateVectorType(TypeVector *type) {
@@ -773,9 +769,9 @@ DIType DIBuilder::CreateTypeDescription(Type *t, bool voidToUbyte) {
     return nullptr;
   if (t->ty == TY::Tnull) {
     // display null as void*
-    return DBuilder.createPointerType(CreateTypeDescription(Type::tvoid), 8, 8,
-                                      /* DWARFAddressSpace */ llvm::None,
-                                      "typeof(null)");
+    return DBuilder.createPointerType(
+        CreateTypeDescription(Type::tvoid), target.ptrsize * 8, 0,
+        /* DWARFAddressSpace */ llvm::None, "typeof(null)");
   }
   if (auto te = t->isTypeEnum())
     return CreateEnumType(te);
@@ -794,13 +790,11 @@ DIType DIBuilder::CreateTypeDescription(Type *t, bool voidToUbyte) {
   if (t->ty == TY::Tstruct)
     return CreateCompositeType(t);
   if (auto tc = t->isTypeClass()) {
-    LLType *T = DtoType(t);
     const auto aggregateDIType = CreateCompositeType(t);
     const auto name =
         (tc->sym->toPrettyChars(true) + llvm::StringRef("*")).str();
-    return DBuilder.createPointerType(aggregateDIType, getTypeAllocSize(T) * 8,
-                                      getABITypeAlign(T) * 8, llvm::None,
-                                      processDIName(name));
+    return DBuilder.createPointerType(aggregateDIType, target.ptrsize * 8, 0,
+                                      llvm::None, processDIName(name));
   }
   if (auto tf = t->isTypeFunction())
     return CreateFunctionType(tf);

